docker.gradle

```diff
@@ -22,14 +23,21 @@ ext {
     dockerTimeout = project.hasProperty('dockerTimeout') ? dockerTimeout.toInteger() : 840
     dockerRetries = project.hasProperty('dockerRetries') ? dockerRetries.toInteger() : 3
     dockerBinary = project.hasProperty('dockerBinary') ? [dockerBinary] : ['docker']
+    dockerBuildArgs = project.hasProperty('dockerBuildArgs') ? dockerBuildArgs : ''
 }
 
 if(project.hasProperty('dockerHost')) {
     dockerBinary += ['--host', project.dockerHost]
 }
 
 task distDocker << {
-    def cmd = dockerBinary + ['build', '-t', dockerImageName, project.buildscript.sourceFile.getParentFile().getAbsolutePath()]
+    def cmd
+    if (dockerBuildArgs != '') {
+        cmd = dockerBinary + ['build', '--build-arg', dockerBuildArgs, '-t', dockerImageName, project.buildscript.sourceFile.getParentFile().getAbsolutePath()]
+    }
+    else {
+        cmd = dockerBinary + ['build', '-t', dockerImageName, project.buildscript.sourceFile.getParentFile().getAbsolutePath()]
+    }
```

Comment:
You might want to follow the same pattern as with `dockerHost`.  Check if the variable is set (don't define a default) and then create a new variable `dockerBuildArgsArgument` which holds the complete `--build-arg test1 test2`. That prevents us from having to nest the `if`s in the `distDocker` block more and more.